### ISSUE TITLE
x/tools/gopls: 

### ISSUE THREAD

**Issue by cyberbearwrites** [golang/go#52641]
### gopls version

Build info
----------
golang.org/x/tools/gopls v0.8.3
    golang.org/x/tools/gopls@v0.8.3 h1:Mxm94ix8oSARQ6svioO6SxKEYWT/VCP54/448LOHzrk=
    github.com/google/go-cmp@v0.5.7 h1:81/ik6ipDQS2aGcBfIN5dHDB36BwrStyeAQquSYCV4o=
    github.com/sergi/go-diff@v1.1.0 h1:we8PVUC3FE2uYfodKH/nBHMSetSfHDR6scGdBi+erh0=
    golang.org/x/mod@v0.6.0-dev.0.20220106191415-9b9b3d81d5e3 h1:kQgndtyPBW/JIYERgdxfwMYh3AVStj88WQTlNDi2a+o=
    golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c h1:5KslGYwFpkhGh+Q16bwMP3cOontH8FOep7tGV86Y7SQ=
    golang.org/x/sys@v0.0.0-20220209214540-3681064d5158 h1:rm+CHSpPEEW2IsXUib1ThaHIjuBVZjxNgSKmBLFfD4c=
    golang.org/x/text@v0.3.7 h1:olpwvP2KacW1ZWvsR7uQhoyTYvKAupfQrRGBFM352Gk=
    golang.org/x/tools@v0.1.11-0.20220407163324-91bcfb1bdf9c h1:rYiuK/jn6TCxjAq1hQ6MR6pgOeP3A5gIHDxgrQ8Vbys=
    golang.org/x/vuln@v0.0.0-20220324005316-18fd808f5c7f h1:9dMzk88fnONra7zrEalqkRMGa9jMGf9B5mdzhYVyI28=
    golang.org/x/xerrors@v0.0.0-20200804184101-5ec99f83aff1 h1:go1bK/D/BFZV2I8cIQd1NKEZ+0owSTG1fDTci4IqFcE=
    mvdan.cc/gofumpt@v0.3.0 h1:kTojdZo9AcEYbQYhGuLf/zszYthRdhDNDUi2JKTxas4=
    mvdan.cc/xurls/v2@v2.4.0 h1:tzxjVAj+wSBmDcF6zBB7/myTy3gX9xvi8Tyr28AuQgc=
go: go1.14.7

### go env

GO111MODULE="auto"
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/kali/.cache/go-build"
GOENV="/home/kali/.config/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/kali/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/kali/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.18.1"
GCCGO="gccgo"
GOAMD64="v1"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/usr/local/go/src/sync/atomic/go.mod"
GOWORK=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build206143841=/tmp/go-build -gno-record-gcc-switches"

### What did you do?
I am a bit new to the Go Programming Language.
 
I upgraded Go to Version 1.81.

After upgrading, everything works perfectly fine except for this error in my VSCODE  problems pane.

undeclared name: any (compile)

![go](https://user-images.githubusercontent.com/69829368/166146444-ce77d6e1-c782-4bc8-8504-435c4f0d8806.png)

I did a bit of googling and all roads seem to lead to using `go mod init`

### What did you expect to see?
Even after running 

`go mod init atomic`

Hoping to resolve the error, problem still persist

In fact even when I try to run `go mod tidy` from vscode I get this when the file is actually there.
Maybe I'm doing this wrongly.

Command 'gopls.tidy' failed: Error: no go.mod file found in /usr/local/go/src/sync/atomic. 

![go2](https://user-images.githubusercontent.com/69829368/166150828-5cc56cd9-ca87-4f49-91f1-5fa50e1c6a34.png)

### Editor and settings

VSCODE; No setting were made relating to Go.

### Logs

[Error - 15:57:04] 2022/05/01 15:57:04 DocumentSymbols failed: getting file for DocumentSymbols: no packages for non-Go file file:///usr/local/go/src/sync/atomic/go.mod (go.mod)
	URI=file:///usr/local/go/src/sync/atomic/go.mod

EDIT: I've never used or opened this file(s) before, I just installed Go, opened vscode and the error appears.

**Comment by hyangah** [comment]
Can u please run "Go: Install/update Tools" command and rebuild gopls with go1.18.x?

Wasn't there a notification to ask u to rebuild all tools? If not, that is a bug.

**Comment by cyberbearwrites** [comment]
> Can u please run "Go: Install/update Tools" command and rebuild gopls with go1.18.x?
> 
> Wasn't there a notification to ask u to rebuild all tools? If not, that is a bug.

Thanks for the responsive feedback.

There was a notification and I let it run.

Let me force it.

**Comment by cyberbearwrites** [comment]
So I started by deleting the previously built gopls binary in my $GOPATH bin file

```bash 
rm $GOPATH/bin/gopls
```

I also deleted staticcheck where it was previously installed on my computer so I can get a new binary built for Go 1.8.1(Although, I don't think it was necessary).

```bash
rm $GOPATH/bin/staticcheck
```

I also went ahead to delete the source files for gopls(I also think this too wasn't necessary).

```bash
rm -r $GOPATH/src/golang.org/x/tools/gopls
```

By installing gopls manually solved my problem

```bash
go get -v golang.org/x/tools/gopls
```
Although there could have been a better way to do this, this was what worked for me.

Thank you for your time [hyangah](https://github.com/hyangah)
